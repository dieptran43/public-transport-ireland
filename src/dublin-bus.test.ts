'use strict'

import { join } from 'path'
import { readFileSync } from 'fs'
import nock from 'nock'
import dublinBus from './dublin-bus'

const wsdlResponse = readFileSync(join(__dirname, 'fixtures', 'dublin-bus', 'wsdl.xml'))
const allDestinationsResponseData = readFileSync(join(__dirname, 'fixtures', 'dublin-bus', 'get-all-destinations.xml'))
const realTimeInfoData = readFileSync(join(__dirname, 'fixtures', 'dublin-bus', 'get-real-time-stop-data.xml'))

// mock WSDL request
nock('http://rtpi.dublinbus.ie')
  .get('/DublinBusRTPIService.asmx?WSDL')
  .reply(200, wsdlResponse)

// mock get all destinations request
nock('http://rtpi.dublinbus.ie', {
  reqheaders: {
    SOAPAction: '"http://dublinbus.ie/GetAllDestinations"'
  }
})
  .post('/DublinBusRTPIService.asmx')
  .once()
  .reply(200, allDestinationsResponseData)

// mock get real time info request
nock('http://rtpi.dublinbus.ie', {
  reqheaders: {
    SOAPAction: '"http://dublinbus.ie/GetRealTimeStopData"'
  }
})
  .post('/DublinBusRTPIService.asmx')
  .once()
  .reply(200, realTimeInfoData)

test('It should get all stops', async function () {
  const stops = await dublinBus.getStops()
  expect(stops).toMatchSnapshot()
})

test('It should get real time information for a stop', async function () {
  const realTimeInfo = await dublinBus.getRealTimeInfo(22)
  expect(realTimeInfo).toMatchSnapshot()
})
