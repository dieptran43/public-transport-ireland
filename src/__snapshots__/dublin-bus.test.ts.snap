// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It should get all stops 1`] = `
Array [
  Object {
    "code": "22",
    "description": "Ballymun Road, Nursing Home",
    "id": 22,
    "latitude": 53.399092,
    "longitude": -6.264086,
  },
  Object {
    "code": "17",
    "description": "Ballymun Road, Nursing Home",
    "id": 17,
    "latitude": 53.399107,
    "longitude": -6.263668,
  },
]
`;

exports[`It should get real time information for a stop 1`] = `
Array [
  Object {
    "arrivingInMinutes": 0,
    "destination": "11 - St Pappin's Rd via Drumcondra",
    "destinationName": "St Pappin's Rd via Drumcondra",
    "expectedArrivalTime": "2019-09-21T17:21:33.927+01:00",
    "lineName": "11",
    "vehicleAtStop": true,
  },
  Object {
    "arrivingInMinutes": 35,
    "destination": "11 - St Pappin's Rd via Drumcondra",
    "destinationName": "St Pappin's Rd via Drumcondra",
    "expectedArrivalTime": "2019-09-21T17:55:51.000+01:00",
    "lineName": "11",
    "vehicleAtStop": false,
  },
  Object {
    "arrivingInMinutes": 59,
    "destination": "11 - St Pappin's Rd via Drumcondra",
    "destinationName": "St Pappin's Rd via Drumcondra",
    "expectedArrivalTime": "2019-09-21T18:20:01.000+01:00",
    "lineName": "11",
    "vehicleAtStop": false,
  },
]
`;
